### PR TITLE
[Cython Build] Pass all compile definitions

### PR DESCRIPTION
**Pull request by Devjiu**
This pr passes all compile definitions and include dirs from CMake to cython build.
Also removing dependencies to cuda from library headers.

Resolves: https://github.com/intel-ai/hdk/issues/297

Signed-off-by: Dmitrii Makarenko <dmitrii.makarenko@intel.com>

### CODE REVIEW

**Comment by Devjiu** [comment]
As currently I am changing headers like `Bakend.h`, `Execute.h` etc. Adding @kurapov-peter to check, that I am not breaking L0 and current changes fine.

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,10 @@
 set(SETUP_PY "${CMAKE_CURRENT_BINARY_DIR}/setup.py")
 
+get_property(CDEFS DIRECTORY PROPERTY COMPILE_DEFINITIONS)
+get_property(IDIRS DIRECTORY PROPERTY INCLUDE_DIRECTORIES)
+message(STATUS "Defintions to Cython: ${CDEFS}")
+message(STATUS "Include dirs to Cython:  ${IDIRS}")
```
Comment (Devjiu) [inline python/CMakeLists.txt:6]: Should I keep the messages or remove them?

File: omniscidb/QueryEngine/Compiler/Backend.h [hunk 2]
```diff
@@ -165,7 +171,7 @@ class CUDABackend : public Backend {
                                       llvm::TargetMachine* nvptx_target_machine);
 
   static std::unique_ptr<llvm::TargetMachine> initializeNVPTXBackend(
-      const CudaMgr_Namespace::NvidiaDeviceArch arch);
+      const CudaMgr_Namespace::NvidiaDeviceArch varch);
```
Comment (ienkovich) [inline omniscidb/QueryEngine/Compiler/Backend.h:174]: Spurious change?

File: python/CMakeLists.txt [hunk 3]
```diff
@@ -1,5 +1,10 @@
 set(SETUP_PY "${CMAKE_CURRENT_BINARY_DIR}/setup.py")
 
+get_property(CDEFS DIRECTORY PROPERTY COMPILE_DEFINITIONS)
+get_property(IDIRS DIRECTORY PROPERTY INCLUDE_DIRECTORIES)
+message(STATUS "Defintions to Cython: ${CDEFS}")
```
Comment (ienkovich) [inline python/CMakeLists.txt:5]: Please remove debug prints.

**Review by ienkovich — COMMENTED** [review]
Overall looks good!

File: omniscidb/QueryEngine/Compiler/Backend.h (hunk 2 above)
Comment (Devjiu) [reply omniscidb/QueryEngine/Compiler/Backend.h:174]: yes, thanks for reminder)

Comment (Devjiu) [reply omniscidb/QueryEngine/Compiler/Backend.h]: resolved

File: python/CMakeLists.txt (hunk 3 above)
Comment (Devjiu) [reply python/CMakeLists.txt]: removed.

**Comment by Devjiu** [comment]
Note:

Only libraries parameter for all `*.pyx` remains unchanged. Also can be modified, if libraries can be taken from some specific targets.

**Review by ienkovich — APPROVED** [review]

**Pull request merged**